### DISCUSSION TITLE
Upgrading from 1.x to 2.x with python binding

### DISCUSSION THREAD

**Discussion by christallire** [cdk8s-team/cdk8s-plus#1689] [Q&A]
Hey guys, I've been trying to upgrade cdk8s plus project with latest version of package due to new functionalities (e.g tolerate)

The problem I'm having is, many of binded function prototype is changed to supply positional arguments like below:
```
serviceAccount = kplus.ServiceAccount.from_service_account_name('docker-pull')
TypeError: ServiceAccount.from_service_account_name() missing 2 required positional arguments: 'id' and 'name'
```

Could you please confirm this change is intended?
I've checked through released packages and 2.0.0-b11 doesn't require these parameters but others up and higher is.

Thanks!

**Comment by iliapolo** [comment]
@christallire Apologies for the (very) delayed response. Yes this was an intentional change we made just before releasing the library out of beta. Here is the PR that introduced it, it contains some reasoning behind the change. 

https://github.com/cdk8s-team/cdk8s-plus/issues/747

Did you manage to eventually upgrade and use the new version?

**Reply by christallire** [reply]
although it worked - It feels super redundant because id doesn't used anywhere in the manifest, for example:

```
        secret1 = kplus.Secret.from_secret_name(self, 'secret-appsettings', 'secret-appsettings')
        secret2 = kplus.Secret.from_secret_name(self, 'secret-signing', 'secret-signing')
        volume1 = kplus.Volume.from_secret(self, 'secrets1', secret1, name='secrets1')
        volume2 = kplus.Volume.from_secret(self, 'secrets2', secret2, name='secrets2')
        
                'ELASTIC_APM_ENVIRONMENT': kplus.EnvValue.from_secret_value(secret_value=kplus.SecretValue(key='ELASTIC_APM_ENVIRONMENT', secret=kplus.Secret.from_secret_name(self, 'elastic-apm-secret1', 'elastic-apm-secret'))),
                'ELASTIC_APM_SECRET_TOKEN': kplus.EnvValue.from_secret_value(secret_value=kplus.SecretValue(key='ELASTIC_APM_SECRET_TOKEN', secret=kplus.Secret.from_secret_name(self, 'elastic-apm-secret2', 'elastic-apm-secret'))),
                'ELASTIC_APM_SERVER_URL': kplus.EnvValue.from_secret_value(secret_value=kplus.SecretValue(key='ELASTIC_APM_SERVER_URL', secret=kplus.Secret.from_secret_name(self, 'elastic-apm-secret3', 'elastic-apm-secret')))
```

Things I noticed during migration:
1) we can't use duplicated id for this purpose: I had to name it by incrementing postfix everywhere when I used it, even if it's the same resource
2) super redundant. I understand why you need it but we shouldn't mitigate the burden on everywhere - especially for utility/helper functions (e.g: Secret.from_secret_name)
  - it would be nicer if we kept the prototype :)

**Comment by christallire** [comment]
YES! it worked!
thkanks
